src/posts/comparing-methods-for-accessing-secrets-in-vault-from-kubernetes.md

```diff
@@ -0,0 +1,285 @@
+
+#### Effort
+
+**Platform Effort**: (Assuming you already have ArgoCD) This requires [installing the plugin](https://argocd-vault-plugin.readthedocs.io/en/stable/installation/) for which the [traditional approach is deprecated](https://argo-cd.readthedocs.io/en/stable/operator-manual/config-management-plugins/#configmap-plugin) and the newer [sidecar-based approach](https://argo-cd.readthedocs.io/en/stable/operator-manual/config-management-plugins/#sidecar-plugin) is recommended, and much more maintainable.
+
+**Deployment Effort**: Writing the manifests can be a bit clumsy to get right but the ArgoCD UI makes it quite clear when something did not work, which helps a lot. And because you are templating raw Kubernetes manifests, it is also very easy to check that everything went as intended.
+
+**Application Effort**: The application can read from environment variables or files.
+
+|  | Platform Effort | Deployment Effort | Application Effort |
+| --- | --- | --- | --- |
+| ArgoCD Vault Plugin | ■■■□□ | ■■□□□ | ■□□□□ |
+
+#### Support
+
+The ArgoCD Vault Plugins support is very similar to that of the External Secrets Operator. So many of the same points/questions apply.
+
+|  | Secrets Engines | Auth Methods | Sync to K8s Secrets | Secret Rotation |
+| --- | --- | --- | --- | --- |
+| ArgoCD Vault Plugin | KV only | AppRole, Token, Github, Kubernetes and Userpass | Optional | Manual |
+
+## Wrapping up
+
+When considering the effort involved in the different approaches, it is important to consider your context. Are you a small, highly connected team, or are you a big enterprise with dedicated platform teams? In the former case, perhaps it doesn’t matter so much where the effort is placed. But in the latter case, it should be a goal to pay the effort on the platform team as much as possible to enable stream-aligned teams to securely access their secrets with minimal effort. After all, there will be many more stream-aligned teams than platform teams so the approach should be favourably scalable.
+
+|  | Platform Effort | Deployment Effort | Application Effort |
+| --- | --- | --- | --- |
+| Vault REST API | ■□□□□ | ■■□□□ | ■■■■■ |
+| Vault Sidecar Injector | ■■□□□ | ■■■■■ | ■■□□□ |
+| Vault CSI Driver | ■■■□□ | ■■■■□ | ■□□□□ |
+| External Secrets Operator | ■■□□□ | ■■■□□ | ■□□□□ |
+| ArgoCD Vault Plugin | ■■■□□ | ■■□□□ | ■□□□□ |
+
+When considering the features of the various approaches, a few things we encountered that helped guide the decision:
+
+1. **Third party tools**. If you manage a lot of 3rd party tools on Kubernetes, you might find it quite painful to try and use an approach that does not support synchronising with Kubernetes secrets. This is because not all tools support sane approaches for reading secrets, or if you rely on Helm, the Helm chart might not work as you intend or differ greatly from tool to tool.
```

Comment:
```suggestion
1. **Third party tools**. If you manage a lot of 3rd party tools on Kubernetes, you might find it quite painful to try and use an approach that does not support synchronising with Kubernetes secrets. This is because not all tools support sane approaches for reading secrets. If you rely on Helm, the Helm chart might not work as you intend or differ greatly from tool to tool.
```